enumerable.rb

```diff
@@ -0,0 +1,257 @@
+# rubocop:disable Style/CaseEquality, Style/For, Lint/RedundantCopDisableDirective, Lint/MissingCopEnableDirective, Metrics/ModuleLength, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    arr = self
+    if block_given?
+      for i in arr do
+        yield i
+      end
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_each_with_index
+    arr = self
+    if block_given?
+      j = 0
+      for i in arr do
+        yield i, j
+        j += 1
+      end
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      arr = self
+      arr.my_each do |i|
+        new_arr.push(i) if yield i
+      end
+      new_arr
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_all?(all_arg = nil)
+    arr = self
```

Comment:
* [ ] There should be a working `my_all?` dentical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
check the example below, they do not return same answers
```p [1, 2i, 3.14].my_all?(Numeric)```  != ```p [1, 2i, 3.14].all?(Numeric)```